### PR TITLE
feat: Support resource reservations on Docker compose import

### PR DESCRIPTION
**Pull request by victorcolombo**
## Description:

This PR adds support to converting `reservations: cpu` field in Docker Compose into `min_cpu` in ServiceConfig. After this change, users will be able to leverage Starlark Validator to get warnings about if CPU/Memory use.

## Is this change user facing?
YES

### CODE REVIEW

File: cli/cli/commands/import/import.go [hunk 1]
```diff
@@ -241,6 +251,36 @@ func convertComposeProjectToStarlark(compose *types.Project) (string, map[string
 	return script, requiredFileUploads, nil
 }
 
+func getMemoryMegabytesReservation(deployConfig *types.DeployConfig) int {
+	if deployConfig == nil {
+		return 0
+	}
+	reservation := 0
+	if deployConfig.Resources.Reservations != nil {
```
Comment (h4ck3rk3y) [inline cli/cli/commands/import/import.go:259]: is `Resources` bound to be set or can `Resources` be nil?

File: cli/cli/commands/import/import.go [hunk 2]
```diff
@@ -241,6 +251,36 @@ func convertComposeProjectToStarlark(compose *types.Project) (string, map[string
...
+	}
+	return reservation
+}
+
+func getMilliCpusReservation(deployConfig *types.DeployConfig) int {
+	if deployConfig == nil {
+		return 0
+	}
+	reservation := 0
+	if deployConfig.Resources.Reservations != nil {
```
Comment (h4ck3rk3y) [inline cli/cli/commands/import/import.go:271]: similar is resources bound to be set?
if resources is set is nanocpus definitely set? (probably as its a float?)

**Review by h4ck3rk3y — APPROVED** [review]

File: cli/cli/commands/import/import.go (hunk 1 above)
Comment (h4ck3rk3y) [reply cli/cli/commands/import/import.go:259]: lets add a test case for this?

Comment (victorcolombo) [reply cli/cli/commands/import/import.go:259]: It is a value not a pointer, so per type system it must be set!

**Pull request merged**